deploy/before_symlink.rb

```diff
@@ -7,13 +7,17 @@
   group 'www-data'
 end
 
-# Pre-compile assets. Also, a very small subset of the assets
-# need to be available without digest fingerprints in their
-# filenames - assets that live "in the wild" and can't be
-# updated whenever our asset fingerprints change.
+execute 'Install JavaScript dependencies and pre-compile assets' do
+  cwd release_path
+  environment NODE_ENV: 'production'
+  command "sudo su search -c 'yarn install --production && RAILS_ENV=#{rails_env} bundle exec rake assets:precompile'"
```

Comment:
JavaScript dependencies are being installed through `yarn install` in this file instead of the cookbooks. As far as I can tell, we'd have to either modify the internals of opsworks_ruby (specifically `deploy[usasearch] action deploy (opsworks_ruby::deploy line 24)`) or modify the order of operations in `before_symlink.rb` in order to run the yarn command in the cookbooks.

Reply:
@ddzz , talk to Rob about that. The installation would just need to happen before `opsworks_ruby` runs its deployment code. The [usasearch/recipes/deploy.rb](https://github.com/GSA/usasearch-cookbooks/blob/main/usasearch/recipes/deploy.rb) file contains our custom deployment code. There are several other packages that search-gov depends on that we install via the cookbooks, so there is precedent for the type of installation we'll need to do for `yarn`.  Take a look at:
https://github.com/GSA/usasearch-cookbooks/blob/7ed524ab7d790275095b103cf12d888e29f96f21/usasearch/recipes/packages.rb
https://github.com/GSA/usasearch-cookbooks/blob/main/usasearch/recipes/dependencies.rb

Reply:
It can be done through the cookbooks, and I think it should be done that way, but we'd have to take out the `assets:precompile` from `before_symlink.rb` and run it later during the deploy, which would require its own PR. I think we should keep this as-is for this PR and address this in a future task.

Running `yarn install` is different than the other packages installed through the cookbooks since `search-gov` needs to be cloned first and `yarn install` needs to be run in the repo because it needs the `package.json` in this PR. Right now, `before_symlink` runs immediately after the repo is cloned, and there's no obvious way to add the `yarn install` in between them other than modifying opsworks_ruby itself.

Reply:
OK - just leave a comment including the follow-up ticket ID to see if we can move that to the cookbooks.